src/components/views/article/Article.js

```diff
@@ -0,0 +1,122 @@
+              </div>
+              <div id="author__profile__div">
+                <h5 className="author__name">{singleArticle?.author?.firstName} {singleArticle?.author?.lastName}</h5>
+                <button className="follow__button" type="button">Follow</button>
+              </div>
+            </div>
+            <div className="article__content">
+              <p className="article__read__time">{singleArticle.readTime} Min Read</p>
+              <h1 className="article__title">{singleArticle.title}</h1>
+              <p className="article__description">
+                {singleArticle.description}
+              </p>
+              <div className="article__body">
+                {Object.keys(singleArticle).length > 0
+                  ? ReactHtmlParser(parseDataFromJSON((JSON.parse(singleArticle.body))))
+                  : <h1>Something went wrong..</h1>
+           }
+              </div>
+              <div className="article__tags">
+                {singleArticle?.tagsList?.length > 0 ? singleArticle.tagsList.map((tag, i) => (
+                  <Tag tagName={tag.toString()} key={i.toString()} />
+                )) : <p className="article__no__tags">No Tags for this article</p>}
+              </div>
+              <div className="article__divider" />
+              <div className="article__actions">
+                <div className="article__ratings">
+                  <i className="material-icons">star_border</i>
+                  <i className="material-icons">star_border</i>
+                  <i className="material-icons">star_border</i>
+                  <i className="material-icons">star_border</i>
+                  <i className="material-icons">star_border</i>
+                </div>
+                <div className="article__interaction">
+                  <i className="material-icons">bookmark_border</i>
+                  <i className="material-icons">share</i>
+                  <i className="material-icons">report</i>
+                </div>
+              </div>
+              <ArticleComment />
+              <div className="article__related__articles">
+                <h3>Related Articles</h3>
+              </div>
+            </div>
+          </div>
+        )
+        }
+      </div>
+    </Fragment>
+  );
+};
+
+Article.propTypes = {
+  getArticleBySlug: PropTypes.func.isRequired,
+  match: PropTypes.shape({}).isRequired,
+  singleArticle: PropTypes.shape({}),
+  error: PropTypes.shape({}),
+  loading: PropTypes.bool.isRequired,
+};
+
+Article.defaultProps = {
+  singleArticle: {},
+  error: {},
+};
+
+const mapStateToProps = state => ({
+  singleArticle: state.singleArticle.article,
+  loading: state.singleArticle.loading,
+  error: state.singleArticle.error,
+});
+
+export const ArticleComponent = Article;
+
+export default connect(
+  mapStateToProps,
+  { getArticleBySlug },
+)(Article);
```

Comment:
I love how you think of edge cases and handle them accordingly, well done. 👍🏽

Reply:
Thank you